hpx/lcos/local/detail/sliding_semaphore.hpp

```diff
@@ -35,6 +35,13 @@ namespace hpx { namespace lcos { namespace local { namespace detail
           : max_difference_(max_difference), lower_limit_(lower_limit), cond_()
         {}
 
+        void set_max_difference(std::int64_t max_difference,
```

Comment:
Is it safe to call this at any time? And is just `set` a better name? Just because it sets both `max_difference` and `lower_limit`. Looks useful though.

Reply:
The max diff and lower limit don't change constantly (unlike the internal counter), so changing them will not be a big deal, it might happen that you change it, whilst a task is completing and one extra or less is alowed through. for the kind of stuff we're using it for, this isn't worth putting locks on. but I'll bear it in mind next time I tweak it.
I can change the name to ust `set`, but then I'd have to go through all my code and change that too, so I'd rather just leave it. Once I've transitioned the cholesky code to use limiting executor, I' can tweak this then if it makes you happy.

Note that it turns out that the limiting execuitor isn't a simple drop-in replacement in some code, because the guided executor has template issues when I try to use it with the limiting one, so I'm working on that.

Reply:
Of course it makes me happy. That's what I live for.